### PR TITLE
Adds applesauce as a fluid [refs #12]

### PR DESCRIPTION
**Pull request by ladenedge**
After much difficulty figuring out how stupid fluids work -- they seem a little overly complicated?!? -- I believe I have a pretty clean implementation for applesauce.

First of all, the fluid is only really available when the Create and Create Central Kitchen mods are present, for two reasons:

1. All of the recipes require Create tools (such as mixing, emptying, etc).
2. I've removed the fluid from JEI similar to Create's Melon Juice, Apple Cider, etc.  In other words, it's a "recipe-only fluid" that doesn't have a bucket or a block.

Here's a shot of JEI looking for "applesauce":

image

Only the food item is present.  Moreover, only the standard cooking recipe is available:

image

However, if you uncomment these lines in **build.gradle**, you can see the fluid in action:

```gradle
// These are helpful when testing Create interoperability.
// runtimeOnly fg.deobf("curse.maven:create-328085:4835191")
// runtimeOnly fg.deobf("curse.maven:create-central-kitchen-820977:5132704")
```

Now we have a new way to create applesauce bowls:

image

And we have ways of creating the fluid from the bowl:

image

As well as a few recipes that can be made directly with the fluid:

image

Here's a shot of the fluid traveling through a pipe after being emptied:

image

And here it is in a big vat:

image

(It's hard to see there, but it is slightly translucent.)

Let me know what you think -- here or on Discord -- and please don't hesitate to suggest even the smallest changes!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,6 @@ run
 
 # Files from Forge MDK
 forge*changelog.txt
+
+# image editing
+*.xcf
```
Comment (ladenedge) [inline .gitignore:28]: I used GIMP to edit images, so it's kind of nice to just leave those files in the same directory..?

File: build.gradle [hunk 2]
```diff
@@ -157,9 +157,13 @@ dependencies {
 
     // Example mod dependency with JEI - using fg.deobf() ensures the dependency is remapped to your development mappings
     // The JEI API is declared for compile time use, while the full JEI artifact is used at runtime
-    // compileOnly fg.deobf("mezz.jei:jei-${mc_version}-common-api:${jei_version}")
-    // compileOnly fg.deobf("mezz.jei:jei-${mc_version}-forge-api:${jei_version}")
-    runtimeOnly fg.deobf("mezz.jei:jei-1.20.1-forge:15.2.0.27")
+    compileOnly fg.deobf("mezz.jei:jei-${minecraft_version}-common-api:${jei_version}")
+    compileOnly fg.deobf("mezz.jei:jei-${minecraft_version}-forge-api:${jei_version}")
+    runtimeOnly fg.deobf("mezz.jei:jei-${minecraft_version}-forge:${jei_version}")
```
Comment (ladenedge) [inline build.gradle:162]: These are all in there for the JEI plugin that blacklists the applesauce fluid.  (To make it a "recipe fluid," similar to Melon Juice and Apple Cider.)

File: build.gradle [hunk 3]
```diff
@@ -157,9 +157,13 @@ dependencies {
...
+
+    // These are helpful when testing Create interoperability.
+    // runtimeOnly fg.deobf("curse.maven:create-328085:4835191")
+    // runtimeOnly fg.deobf("curse.maven:create-central-kitchen-820977:5132704")
```
Comment (ladenedge) [inline build.gradle:166]: These are handy for testing the Create integration.

File: src/main/java/de/chrisimo/vegandelight/compat/jei/VeganDelightJeiPlugin.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    /**
+     * Removes recipe-only items/fluids from the JEI interface.
+     *
+     * @param registration JEI's recipe registration.
+     */
+    @Override
+    public void registerRecipes(IRecipeRegistration registration) {
+        var stacks = RecipeFluid.RECIPE_FLUIDS.stream().map(f -> new FluidStack(f, 1000)).toList();
+        registration.getIngredientManager().removeIngredientsAtRuntime(ForgeTypes.FLUID_STACK, stacks);
```
Comment (ladenedge) [inline src/main/java/de/chrisimo/vegandelight/compat/jei/VeganDelightJeiPlugin.java:42]: Here we are blacklisting all the "recipe fluids," which are just meant to be fluids that you can't dump and swim in and so forth.

**Comment by SayWhatSayMon** [comment]
@ladenedge I love your detailed documentation! Good job.

**Pull request merged**